apps/angular/src/app/features/anime-table/anime-table.component.ts

```diff
@@ -46,6 +109,138 @@ export class AnimeTableComponent {
 	/** Column key. */
 	protected readonly columnKey = ColumnKey;
 
+	/** Data source. */
+	protected animeList$ = new Observable<readonly Anime[]>();
+
+	/** Page size. */
+	protected readonly pageSizes = [25, 50, 100];
+
+	/** Total count. */
+	protected totalCount = 0;
+
 	/** Date format. */
 	protected readonly dateFormat = DATE_FORMAT;
+
+	/** Sortable fields. */
+	protected readonly sortableFields = [ColumnKey.TitleEnglish, ColumnKey.AiredStart, ColumnKey.Status];
+
+	@ViewChild(MatPaginator)
+	private readonly paginator!: MatPaginator;
+
+	@ViewChild(MatSort)
+	private readonly sort!: MatSort;
+
+	private getAnimeList(params: AnimeManagementParams): Observable<Pagination<Anime>> {
+		this.navigationService.navigate('', AnimeManagementParamsMapper.toDto(params));
+		return this.animeService.getPaginatedAnime(params);
+	}
+
+	private getQueryParams(): void {
+		this.activatedRoute.queryParams
+			.pipe(
+				tap(value => {
+					this.pageSize$.next(value['limit'] ?? INITIAL_PAGE_SIZE);
+					this.pageNumber$.next(Math.round((value['offset'] ?? 0) / (this.getSubjectValue(this.pageSize$) ?? INITIAL_PAGE_SIZE)));
+					this.search = value['search'] ?? '';
+					this.ordering$.next(value['ordering']);
+					this.filter = value['type__in']?.split(',').map((type: AnimeTypeDto) => AnimeTypeMapper.fromDto(type)) ?? [];
+				}),
+				takeUntilDestroyed(this.destroyRef),
+			)
+			.subscribe();
+	}
+
+	private getSubjectValue<T>(subject$: BehaviorSubject<T>): T | null {
+		let subjectValue: T | null = null;
+		subject$.pipe(take(1)).subscribe(value => {
+			subjectValue = value;
+		});
+		return subjectValue;
+	}
+
+	private subscribeToControls(): void {
+		this.paginator.page.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(() => {
+			this.pageNumber$.next(this.paginator.pageIndex);
+			this.pageSize$.next(this.paginator.pageSize);
+		});
+
+		this.sort.sortChange
+			.pipe(takeUntilDestroyed(this.destroyRef))
+			.subscribe(() => this.ordering$.next(SortParamsMapper.toDto(this.sort.active as SortParams, this.sort.direction)));
+	}
+
+	/** @inheritdoc */
+	public ngOnInit(): void {
+		this.getQueryParams();
+	}
+
+	/** @inheritdoc */
+	public ngOnChanges(changes: SimpleChanges): void {
+		const previousSearchValue = changes['search']?.previousValue;
+		const currentSearchValue = changes['search']?.currentValue;
+		if (previousSearchValue !== currentSearchValue) {
+			this.search$.next(currentSearchValue);
+			this.pageNumber$.next(0);
+		}
+		const previousFilterValue = changes['filter']?.previousValue;
+		const currentFilterValue = changes['filter']?.currentValue;
+		if (this.isFiltersChange(previousFilterValue, currentFilterValue)) {
+			this.filter$.next(currentFilterValue);
+			this.pageNumber$.next(0);
+		}
+	}
+
+	private isFiltersChange(previous: string[] | null, current: string[] | null): boolean {
```

Comment:
It's better to use `readonly` array in function arguments as our wiki [recommends](https://wiki.saritasa.rocks/frontend/languages/ts-js/best-practices#input):
```suggestion
	private isFiltersChange(previous: readonly string[] | null, current: readonly string[] | null): boolean {
```